### ISSUE TITLE
Cannot replicate the main result from your paper

### ISSUE THREAD

**Issue by jovana-mitrovic** [facebookresearch/swav#60]
Hey, I have reimplemented Swav (following your code here) and have run experiments according to the details in the paper.
I've been able to replicate the results you report in figure 3 right (number of epochs vs top-1 accuracy) for 100, 200 and 400 epochs, but cannot replicate your result for 800 epoch which is also the main result of your paper. For 800 epochs, I get worse results than for 400 epochs with the same hyperparameters! 

Can you please tell me if you changed something in your experiments for 800 epochs? Have you seen this behaviour previously?

Thank you.

**Comment by mathildecaron31** [comment]
Hi @jovana-mitrovic 

Thanks for your interest. I've used the exact same set of hyperparameters for the 100, 200, 400 and 800 epochs runs.

When training with mixed precision, I've observed some variance in the runs.

**Comment by jovana-mitrovic** [comment]
Hey @mathildecaron31 

Thanks for getting back to me on this.
How big was the variance that you observed in the long runs?
I run 10 different seeds and still get more than 1% lower accuracy on 800 epochs than what is in the paper (while matching your results on 100, 200 and 400 epochs from the paper).
Did you by any chance observe any other important design choices that made a difference in the long runs, but not in the short runs?

Btw, how important was the fact that you are not using global syncing for batch norm with apex?

**Comment by mathildecaron31** [comment]
Hi @jovana-mitrovic 

Here are the results of 6 independent runs for 800 epochs to give you an idea of the variance I've observed when training with the exact config specified in this repo:
74.9, 75.3, 75.0, 74.6, 75.1, 75.2 

When switching off mixed precision training the runs were more stable. I attach to this reply the loss curve for a 800 epochs training in full precision (which gives 75.2 in linear evaluation).

Not using global syncing was a game changer for speed of training. See my reply here https://github.com/facebookresearch/swav/issues/22#issuecomment-688260926
You need to ensure that the group of processes that synchronize the batch norm is large enough (in my experiments I used 8 processes per group which I think is enough on ImageNet but not enough for datasets that exhibit higher variance) otherwise the performance significantly degrades.
![image](https://user-images.githubusercontent.com/20599397/114990120-7d56f300-9e98-11eb-8776-813c2e0a559f.png)
